src/mmw/js/src/compare/views.js

```diff
@@ -558,6 +574,72 @@ var BarChartRowView = Marionette.ItemView.extend({
     },
 });
 
+var LineChartRowView = Marionette.ItemView.extend({
+    models: models.LineChartRowModel,
+    className: 'compare-chart-row -line',
+    template: compareLineChartRowTmpl,
+
+    ui: function() {
+        return {
+            chartElement: '#' + this.model.get('chartDiv'),
+        };
+    },
+
+    onAttach: function() {
+        this.destroyChart();
+        this.addChart();
+    },
+
+    onDestroy: function() {
+        this.destroyChart();
+    },
+
+    onRender: function() {
+        this.destroyChart();
+        this.addChart();
+    },
+
+    addChart: function() {
+        var scenarioNames = this.model.get('scenarioNames'),
+            chartData = this.model.get('data')
+                .reduce(function(accumulator, scenarioData, index) {
```

Comment:
`map` is clearer and more compact than `reduce`, especially when the output is expected to be an array of the same size as the input:

```js
var chartData = this.model.get('data')
        .map(function(values, index) {
            return {
                key: index,
                values: values.map(function(value, index2) {
                    return {x: index2, y: Number(value)};
                }),
                color: SCENARIO_COLORS[index % 32],
            };
        }).reverse();
```

Reply:
Sounds good to me.